src/daemon/SeatManager.cpp

```diff
@@ -59,6 +60,12 @@ namespace SDDM {
         QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(reply);
         connect(watcher, &QDBusPendingCallWatcher::finished, this, [=]() {
             watcher->deleteLater();
+            if (Logind::serviceName().contains(QStringLiteral("ConsoleKit"))) {
```

Comment:
If we're going to do this, we should move it earlier and skip
 QDBusPendingReply<QVariant> reply = QDBusConnection::systemBus().asyncCall(canGraphicalMsg);

Reply:
I tried that earlier, and the problem is that it needs time for `logindSeatAdded` to watch the property.  Otherwise, the main thread never sees the property change as there is nothing listening for that signal's emission.

Look around [src/daemon/SeatManager.cpp:155](https://github.com/sddm/sddm/blob/develop/src/daemon/SeatManager.cpp#L155).

Reply:
Then change SeatManager.cpp